test/test_query.cpp

```diff
@@ -4125,6 +4155,425 @@ TEST(Query_LinkChainSortErrors)
+            CHECK_EQUAL(tv.get_int(t1_int_col, i), results[i]);
+        }
+    }
+}
+
+
+TEST(Query_DistinctAndSort)
+{
+    Group g;
+    TableRef t1 = g.add_table("t1");
+    TableRef t2 = g.add_table("t2");
+    size_t t1_int_col = t1->add_column(type_Int, "t1_int");
+    size_t t1_str_col = t1->add_column(type_String, "t1_str");
+    size_t t1_link_col = t1->add_column_link(type_Link, "t1_link_t2", *t2);
+    size_t t2_int_col = t2->add_column(type_Int, "t2_int");
+    t1->add_empty_row(6);
+    t2->add_empty_row(6);
+
+    t1->set_int(0, 0, 1); t1->set_string(1, 0, "A"); t1->set_link(2, 0, 1);
+    t1->set_int(0, 1, 1); t1->set_string(1, 1, "A"); t1->set_link(2, 1, 0);
+    t1->set_int(0, 2, 1); t1->set_string(1, 2, "B"); t1->set_link(2, 2, 2);
+    t1->set_int(0, 3, 2); t1->set_string(1, 3, "B"); t1->set_link(2, 3, 3);
+    t1->set_int(0, 4, 2); t1->set_string(1, 4, "A"); t1->set_link(2, 4, 5);
+    t1->set_int(0, 5, 2); t1->set_string(1, 5, "A"); t1->set_link(2, 5, 4);
+
+    t2->set_int(0, 0, 0);
+    t2->set_int(0, 1, 0);
+    t2->set_int(0, 2, 1);
+    t2->set_int(0, 3, 1);
+    t2->set_int(0, 4, 2);
+    t2->set_int(0, 5, 2);
+
+    //     T1                              T2
+    //   | t1_int   t1_str   t1_link_t2  | t2_int  |
+    //   ===========================================
+    // 0 | 1        "A"      1           | 0       |
+    // 1 | 1        "A"      0           | 0       |
+    // 2 | 1        "B"      2           | 1       |
+    // 3 | 2        "B"      3           | 1       |
+    // 4 | 2        "A"      5           | 2       |
+    // 5 | 2        "A"      4           | 2       |
+
+    using ResultList = std::vector<std::pair<size_t, size_t>>; // value, index
+    {   // distinct with no sort keeps original order
+        TableView tv = t1->where().find_all();
+        ResultList expected = {{1, 0}, {2, 3}};
+        tv.distinct(t1_int_col);
+        CHECK_EQUAL(tv.size(), expected.size());
+        for (size_t i = 0; i < tv.size(); ++i) {
+            CHECK_EQUAL(tv.get_int(t1_int_col, i), expected[i].first);
+            CHECK_EQUAL(tv.get_source_ndx(i), expected[i].second);
+        }
+    }
+    {   // distinct on a sorted view retains sorted order
+        TableView tv = t1->where().find_all();
+        ResultList expected = {{1, 0}, {2, 4}};
+        tv.sort(SortDescriptor(*t1, {{t1_str_col}, {t1_int_col}}));
+        tv.distinct(t1_int_col);
+        CHECK_EQUAL(tv.size(), expected.size());
+        for (size_t i = 0; i < tv.size(); ++i) {
+            CHECK_EQUAL(tv.get_int(t1_int_col, i), expected[i].first);
+            CHECK_EQUAL(tv.get_source_ndx(i), expected[i].second);
+        }
+    }
+    {   // distinct on a view sorted descending retains sorted order
+        TableView tv = t1->where().find_all();
+        ResultList expected = {{2, 3}, {1, 2}};
+        bool ascending = false;
+        tv.sort(SortDescriptor(*t1, {{t1_str_col}, {t1_int_col}}, {ascending, ascending}));
```

Comment:
I guess this is not easily solvable. One option could be to write `false /* descending */`as parameter. We should just learn that it would have been better with an enum as parameter.